### DISCUSSION TITLE
Why does useSnapshot always re-render if no properties are accessed?

### DISCUSSION THREAD

**Discussion by dvtng** [pmndrs/valtio#808] [Q&A]
The reason I ask is it makes useProxy and other utils less effective. If I'm understanding correctly, even if a component only mutates the proxy returned from useProxy, that component will re-render on any change due to the useSnapshot call within useProxy?

**Comment by dai-shi** [comment]
It behaves consistently and makes sense for useSnapshot. However, like you said for useProxy it's undesirable. Let me see if we could do a hacky fix.

**Reply by dai-shi** [reply]
#809
